### PR TITLE
Claire Caudwell Happy thoughts API ❤

### PR DESCRIPTION
**Pull request by ClaireCaudwell**
**Link to my deployed API:** https://claireshappythoughts.herokuapp.com/

**Reflections in this week:** Nice project to restart after xmas so very grateful for that. Was really cool to be able to work more with creating a database and endpoints, especially ones that you can use to post data to the database and then actually using in the Happy Thoughts project. 

**Things I want clarified:** Not anything at the moment. I feel like I understand what I am doing and whenever I'm unsure I google it or ask my teammates.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,29 +1,91 @@
...
+  "GET endpoint": {
+    "https://claireshappythoughts.herokuapp.com/thoughts": "Use this endpoint in your application to show up to the last 20 thoughts posted",    
+  },  
+  "POST endpoint 1": {
+    "https://claireshappythoughts.herokuapp.com/thoughts": "Use this endpoint in your application to allow for the user to post a thought to the database. Only allows for the message to be sent in the body of the post request.",
+  },
+  "POST endpoint 2": {
+    "https://claireshappythoughts.herokuapp.com/thoughts/:thoughtId/like": " Use this endpoint to allow for users to like thoughts by including the thought id in the url which targets the thought with that specific id and then updates the heart count of that thought in the database.",
+  },
+};
```
Comment (lilihorvath) [inline server.js:50]: I really like this part of your code, so clean and understandable! Maybe you can separate to different lines to be more readable, but that is just a tip!(:

File: server.js [hunk 2]
```diff
@@ -1,29 +1,91 @@
...
+  // Try catch form, but you can also use promises (then with a catch). This will try to save the new thought and if it doesn't succeed, for instance if it doesn't meet the min and maxlength defined in the model, then the error will be shown.
+  try {
+    // create new thought with the data passed via the body of the post request from the client side via the API endpoint and then save that to the database. Only the message can be updated and not the whole object as message is the only property specified and saved to the database
+    const thought = await new Thought({message: req.body.message}).save();
+    // returning a good status code and the json object that we saved to the database
+    res.status(200).json(thought);
+  } catch (error) {
+    // returning a bad error code and a message with some more information to the user as to why the thought wasn't sent.
+    res.status(400).json({message: "Could not save new thought to the database. Please make sure that you've entered a valid thought that is between 5-140 letters long.", errors:error.errors})
+  }
```
Comment (lilihorvath) [inline server.js:74]: I am not understand how this is work, I am sure that is good, so if you have time, can you explain this for me?(: Great work!

File: README.md [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+6. Then I changed the endpoints in the Happy thoughts project and made sure that they were working and that I was getting the data that was stored on and being posted to my database.
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
-
-## View it live
+## View it live 💌
+Happy thoughts API: https://claireshappythoughts.herokuapp.com/
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+Happy thoughts project: https://silly-wing-fdef06.netlify.app/
```
Comment (lilihorvath) [inline README.md:28]: Incredibly great job Claire! So understandable and clear! I love the way how you write you the read.me files! Congrats!

File: server.js (hunk 2 above)
Comment (ClaireCaudwell) [reply server.js:74]: @lilihorvath Do you meant the error message of the new thought isn't successful?

File: server.js [hunk 4]
```diff
@@ -1,29 +1,91 @@
...
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.json(documentation);
+});
+
+// thoughts get route which will show a max of 20 thoughts, sorted by createdAt to show the most recent 20 thoughts first 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20);
```
Comment (KarolCislowski) [inline server.js:58]: to make your queries string work properly you need to add `.exec()` at the end.

**Review by KarolCislowski — APPROVED** [review]
Nice job 👍🏻

**Pull request closed without merging**